Add nox to run test and environment (#516)
gcovr/gcovr: CONTRIBUTING.rst

Test suite
----------

The QA process (``python3 -m nox``) consists of multiple parts:

- linting and checking format(``python3 -m nox --session lint``)

- tests (``python3 -m nox --session tests``)

   - unit tests in ``gcovr/tests``
   - integration tests in ``gcovr/tests``
   - documentation examples in ``doc/examples``

- documentation build (``python3 -m nox --session doc``)

The tests are in the ``gcovr/tests`` directory.

> You can run the tests with ``python3 -m nox --session tests_version($CC)``.

Author: Yes, this was the intention but we can document `tests` also as the first choice.

Test suite
----------

The QA process (``python3 -m nox``) consists of multiple parts:

- linting and checking format(``python3 -m nox --session lint``)

- tests (``python3 -m nox --session tests``)

   - unit tests in ``gcovr/tests``
   - integration tests in ``gcovr/tests``
   - documentation examples in ``doc/examples``

- documentation build (``python3 -m nox --session doc``)

The tests are in the ``gcovr/tests`` directory.
You can run the tests with ``python3 -m nox --session tests``

> for the default GCC version (used from environment ``CC``iv availabele, else gcc-5.

Reviewer: Typos. I think phrasing it like this is clearer:

```suggestion
for the default GCC version (specified via ``CC`` environment variable, defaults to gcc-5).
```

Run and filter tests
~~~~~~~~~~~~~~~~~~~~

To run all tests, use ``python3 -m nox``.
The tests currently assume that you are using GCC 5
and have set up a :ref:`development environment <development environment>`.
You can select a different GCC version by setting the CC environment variable.
Supported versions are ``CC=gcc-5``, ``CC=gcc-6``, ``CC=gcc-8`` and ``clang-10``.

You can run the tests with additional options by setting the environment variable
``TEST_OPTS``. Run all tests after each change is a bit slow, therefore you can
limit the tests to a specific test file, example project, or output format.
For example:

.. code:: bash

    # run only XML tests
    export TEST_OPTS="-k 'xml'"

>     python3 -m nox --session tests

Author: Yes, I'll update the docs.
